src/Microsoft.AspNetCore.Mvc.Analyzers/AttributesShouldNotBeAppliedToPageModelAnalyzer.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class AttributesShouldNotBeAppliedToPageModelAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(
+            DiagnosticDescriptors.MVC1001_FiltersShouldNotBeAppliedToPageHandlerMethods,
+            DiagnosticDescriptors.MVC1002_RouteAttributesShouldNotBeAppliedToPageHandlerMethods,
+            DiagnosticDescriptors.MVC1003_RouteAttributesShouldNotBeAppliedToPageModels);
+
+        public override void Initialize(AnalysisContext context)
+        {
+#if !DEBUG
+            context.EnableConcurrentExecution();
```

Comment:
Curious, what does this do? Also these warning would be generated both for debug & release builds, right?

Reply:
It causes the analyzers to run in parallel. It ends up being a hassle to debug it on larger code bases. 

That said, I don't see Roslyn doing this for their analyzers so maybe this isn't the best thing to do: https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/CSharp/CSharpImmutableObjectMethodAnalyzer.cs#L48. I'll ask around and are any recommendations for this.

Reply:
Haven't heard back as yet, just doing what everybody else does for now.